src/gtimelog/main.py

```diff
@@ -1,1950 +1,49 @@
-
-    def make_enter_close_the_dialog(self):
-        hb = self.get_header_bar()
-        hb.forall(self._traverse_headerbar_children, None)
-
-    def _traverse_headerbar_children(self, widget, user_data):
-        if isinstance(widget, Gtk.Box):
-            widget.forall(self._traverse_headerbar_children, None)
-        elif isinstance(widget, Gtk.Button):
-            if widget.get_style_context().has_class('close'):
-                widget.set_can_default(True)
-                widget.grab_default()
-
-    def virtual_midnight_changed(self, *args):
-        h, m = self.gsettings.get_value('virtual-midnight')
-        self.virtual_midnight_entry.set_text('{:d}:{:02d}'.format(h, m))
-
-    def virtual_midnight_set(self, *args):
-        try:
-            vm = parse_time(self.virtual_midnight_entry.get_text())
-        except ValueError:
-            self.virtual_midnight_changed()
-        else:
-            h, m = self.gsettings.get_value('virtual-midnight')
-            if (h, m) != (vm.hour, vm.minute):
-                self.gsettings.set_value('virtual-midnight', GLib.Variant('(ii)', (vm.hour, vm.minute)))
-
-    def smtp_port_changed(self, *args):
-        port = self.gsettings.get_int('smtp-port')
-        if port == 0:
-            mail_protocol = self.gsettings.get_string('mail-protocol')
-            default_port = MAIL_PROTOCOLS[mail_protocol].factory.default_port
-            self.port_entry.set_text('auto (%d)' % default_port)
-        else:
-            self.port_entry.set_text(str(port))
-
-    def smtp_port_set(self, *args):
-        port = self.port_entry.get_text()
-        if not port or port.lower().startswith("auto"):
-            port = 0
-        try:
-            port = int(port)
-            if not 0 <= port <= 65535:
-                raise ValueError('value out of range')
-        except ValueError:
-            self.smtp_port_changed()
-        else:
-            self.gsettings.set_int('smtp-port', port)
-
-    def refresh_password_field(self, *args):
-        server = self.gsettings.get_string("smtp-server")
-        username = self.gsettings.get_string("smtp-username")
-
-        def callback(password):
-            # In theory the user could've focused the password field
-            # and started typing in a new password, in which case we shouldn't
-            # overwrite it!
-            self.password_entry.set_text(password)
-
-        if username:
-            start_smtp_password_lookup(server, username, callback)
-        else:
-            self.password_entry.set_text("")
+log = root_logger.getChild('gtimelog')
+mark_time("in script")
 
-    def update_password(self, *args):
-        server = self.gsettings.get_string("smtp-server")
-        username = self.gsettings.get_string("smtp-username")
-        password = self.password_entry.get_text()
-        if username:
-            set_smtp_password(server, username, password)
+if DEBUG:
+    os.environ['G_ENABLE_DIAGNOSTIC'] = '1'
```

Comment:
Doesn't this need to happen before `import gi` if we want it to have any effect?  TBH I don't remember.

Reply:
Hummm yep maybe

Reply:
gi is imported in ui module so this appends before. But configured in main